handout/http/search_with_switchmap.md

```diff
@@ -0,0 +1,91 @@
+# Enhancing Search with `switchMap`
+
+There is a problem with our previous implementation of incremental search.
+
+What if the server, for some reason, takes a very long time to respond to a particular query? If we use `flatMap`, we run the risk of getting results back from the server in the wrong order. Let's illustrate this with an example.
+
+## A Quick Example
+
+Consider a situation where we first type in the letters `ABC`, and suppose the string `ABC` is actually a special string where it will take the server a few extra seconds to reply.
+
+Meanwhile, after we paused for a bit (more than the debounce time), we decide to type in another letter (the letter X) and our app sends a request to the server for the string `ABCX`. Since `ABCX` is not considered a special string, the server replies very quickly and our app sets the suggestions for `ABCX`.
+
+A few seconds later, however, the server finally replies with the response for the `ABC` string, and our app receives that response and sets the search suggestions for `ABC`, overwriting the suggestions for the `ABCX` string, even though the request for that actually came afterwards.
+
+Here is a simple diagram to illustrate the issue:
+
+```
+// A1: Request for `ABC`
+// A2: Response for `ABC`
+// B1: Request for `ABCX`
+// B2: Response for `ABCX`
+
+--A1----------A2-->
+------B1--B2------>
+```
+
+You can see that A2 arrives after B2 even though the A1 request began first. So how do we get around this problem? We can use `switchMap`.
```

Comment:
Proposed change:

You can see that A2 arrives after B2 even though the A1 request began first. This will end up showing the wrong results to the user. "If the last input in the search was `ABCX` why am I seeing the results for `ABC`?" the user might think. To get around this problem we need to replace `flatMap` with `switchMap`.